pkg/os/shell/shell_unix.go

```diff
@@ -16,12 +15,11 @@ var (
 
 // detect detects user's current shell.
 func detect() (string, error) {
-	shell := os.Getenv("SHELL")
-
-	if shell == "" {
+	detectedShell := detectShellByInvokingCommand("", "ps", []string{"-o", "pid,comm"})
```

Comment:
I can see that in that function you are then trying to split the line and ignoring 0 index which can be done using following so we don't have to ignore it later. 

```
$ ps -o pid=,comm=
  57609 bash
 242709 ps
```

Reply:
You're right